frontend/src/types/budget.ts

```diff
@@ -0,0 +1,35 @@
+export interface Transaction {
+  id: number;
+  categoryId: number | null;
+  categoryName: string | null;
+  description: string;
+  withdraw: string | null;
+  deposit: string | null;
+  // TODO: Fix this
+  createDatetime?: Date | string;
+  transactionDatetime: Date | string;
+}
+
+export interface TransactionRequest {
+  categoryId: number;
+  description: string;
+  withdraw: number;
+  deposit: number;
+  transactionDatetime: Date;
+}
+
+export interface TransactionFormItem {
+  categoryId: number;
+  type: 'deposit' | 'withdraw';
+  amount: number;
+  description: string;
+  transactionDatetime: Date;
+}
+
+export interface Category {
```

Comment:
Maybe we will need a separate file for Category-related stuffs and leave Transaction-related stuffs here and renaming this file.

Reply:
yeah, when the file gets long enough, I'm going to do that.